### ISSUE TITLE
Importing large database with source

### ISSUE THREAD

**Issue by howdu** [dbcli/mycli#715]
Importing a database dump via source and nothing is being output until all the queries have been ran, whereas mysql prints the rows affected and time after each query. Is it possible to do the same with mycli?

**Comment by Alexis6307** [comment]
Same request. If you want to load multiple GB databases it is crazy to watch the screen without doing anything for hours.

**Comment by tsroten** [comment]
I can definitely see this as useful. We typically [recommend](https://github.com/dbcli/mycli/issues/663#issuecomment-427574674) using the official client for large dumps as its typically faster.

However, even for small dumps, this feature would be a nice improvement. 👍 

Currently it appears that we [don't segment the file](https://github.com/dbcli/mycli/blob/master/mycli/main.py#L235) into separate queries, we just run the whole thing as a single query (which would explain the lack of output).

**Comment by scottnemes** [comment]
This appears to be working as desired in the latest version; when sourcing a SQL file, it runs queries individually and gives their status as it goes (via sqlexecute.run, which does the split).

```
MySQL scottn@localhost:(none)> source test.sql
Query OK, 1 row affected
Time: 0.008s

Query OK, 1 row affected
Time: 0.002s

Query OK, 1 row affected
Time: 0.002s
MySQL scottn@localhost:(none)>
```

@rolandwalker This should be safe to close.

**Comment by rolandwalker** [comment]
I'm not so sure about this one.  Don't we still read the entire file into memory?

**Comment by scottnemes** [comment]
@rolandwalker Yeah it reads the file into memory and then passes it to sqlexecute.run, which then splits it up into separate queries and runs them one by one. If I'm understanding the original issue it is around the individual query execution results though, not any issues with memory or anything. I.e. it sounds like back then it would execute all the SQL statements before responding back with the individual query results; whereas now at least it does run them one by one and prints the results as far as I can tell anyway.

I ran another test with three queries in a file and sourced it:
SELECT * FROM dev.t1;
SELECT SLEEP(60);
SELECT * FROM dev.t2;

And it runs/returns the results from the first query right away, then sleeps, and then runs/returns the third query as expected. So if the behavior was the same as described originally it seems like it would have waited until all queries were ran (meaning ~60s in this case with the sleep) before returning the results of the first query. But let me know if I'm missing the point there.

**Comment by rolandwalker** [comment]
Then it seems to be fixed.
